Vit2

vit-ef opened the pull request:
- added future paris cohorts
- updated legacy fix for SW/HW to discount old app reviews

vit-ef commented:
- Added future job postings to cohort list
- changed double pipe equals to equals in line 380 so a 'software-hardware-unknown' tag isn't added for the historic app reviews (where the app review is present but the software/hardware question is not)
- changed the search text 'eligible' to 'are they eligible' as there is a question in the coffee form 'Paris Call with Sourced Lead (New)' which has the question 'Should this person have been sourced? (Score 3 and eligible on paper)' which is falsely identified as the eligibility question
- 'cohort they can join'->'cohort they are' to accurately reflect feedback form in lever

andyjy commented on config/rules.rb:
```diff
@@ -169,9 +169,9 @@ def summarise_one_feedback(f, opp)
-        f[:_text].include?('eligible') ||
+        f[:_text].include?('are they eligible') ||
         f[:_text].include?('elligible') ||
-        f[:_text].include?('cohort they can join')
```
are you sure there isn't at least one (legacy?) feedback form with this wording?
i.e. should we keep "cohort they can join" but also add "cohort they are"?

vit-ef replied:
yes good point, will add

andyjy commented on config/rules.rb:
```diff
@@ -376,8 +376,9 @@ def summarise_all_feedback(summaries, opp)
+    
       if result[:has_app_review]
-        result[:app_review_software_hardware] ||= value
+        result[:app_review_software_hardware] = value
```
this change will only affect opportunities with the tag "software" or "hardware" manually added, and mean the presence of one of these tags will override any value from the feedback form, even for the new forms with the soft/hardware question present. 

it will not remove the "unknown" tag for older feedback forms without the soft/hardware question where one of the manual tags has not been added to the opportunity.

is all of the above what we want?

vit-ef replied:
So we asked the teams to add the 'Software' or 'Hardware' tag for those individuals who would have been missed out because of historical app reviews - so the above should cover everyone. 

But now I'm wondering whether we should check for all unknown ones too i.e. if software/hardware is unknown then check other feedback form answers

vit-ef commented:
2020-08-20

- Added: another variation of eligibility question currently found in feedback forms
- Updated: search term used to find eligibility question to rule out false positives
